### ISSUE TITLE
Postgres: Casting to Money is broken

### ISSUE THREAD

**Issue by jonahwh** [metabase/metabase#3754]
- Databases: Postgres 9.5
- Metabase version: 0.20.3
- Metabase hosting environment: AWS
- Metabase internal database: H2

When trying to cast a `decimal` less than `1000` to a money, the dollar sign is stripped off.

psql console:
```
=> SELECT 123.45::money;
  money
---------
 $123.45
(1 row)
```

in metabase: (There is no dollar sign)
![image](https://cloud.githubusercontent.com/assets/794237/20229355/1cd8247a-a813-11e6-9813-c61b27e3f361.png)

When trying to cast a `decimal` GREATER than `999.99` to a money, an exception is thrown

psql console:
```
=> SELECT 1000::money;
   money
-----------
 $1,000.00
(1 row)
```

in metabase:
![image](https://cloud.githubusercontent.com/assets/794237/20229372/44e2c574-a813-11e6-88ed-3fa575556c6c.png)

<details>
<summary>Log + Stacktrace:</summary>

```
Nov 11 13:31:43 stdouterr.log:  PSQLException:
Nov 11 13:31:43 stdouterr.log:   Message: Bad value for type double : 1,000.00
Nov 11 13:31:43 stdouterr.log:   SQLState: 22003
Nov 11 13:31:43 stdouterr.log:   Error Code: 0
Nov 11 13:31:43 stdouterr.log:  11-11 20:31:43 ERROR generic-sql.query-processor :: nil
Nov 11 13:31:43 stdouterr.log:  11-11 20:31:43 ERROR metabase.query-processor :: {:status :failed,
Nov 11 13:31:43 stdouterr.log:   :class java.lang.Exception,
Nov 11 13:31:43 stdouterr.log:   :error "Bad value for type double : 1,000.00",
Nov 11 13:31:43 stdouterr.log:   :stacktrace
Nov 11 13:31:43 stdouterr.log:   ["driver.generic_sql.query_processor$do_with_try_catch.invokeStatic(query_processor.clj:293)"
Nov 11 13:31:43 stdouterr.log:    "driver.generic_sql.query_processor$do_with_try_catch.invoke(query_processor.clj:289)"
Nov 11 13:31:43 stdouterr.log:    "driver.generic_sql.query_processor$execute_query.invokeStatic(query_processor.clj:335)"
Nov 11 13:31:43 stdouterr.log:    "driver.generic_sql.query_processor$execute_query.invoke(query_processor.clj:331)"
Nov 11 13:31:43 stdouterr.log:    "driver$fn__18249$G__18242__18256.invoke(driver.clj:43)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$run_query.invokeStatic(query_processor.clj:490)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$run_query.invoke(query_processor.clj:474)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$pre_check_query_permissions$fn__21808.invoke(query_processor.clj:429)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$limit$fn__21793.invoke(query_processor.clj:402)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$cumulative_count$fn__21788.invoke(query_processor.clj:391)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$cumulative_sum$fn__21758.invoke(query_processor.clj:339)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$post_format_rows$fn__21649.invoke(query_processor.clj:216)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$post_add_row_count_and_status$fn__21594.invoke(query_processor.clj:192)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$wrap_catch_exceptions$fn__21572.invoke(query_processor.clj:122)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$process_query.invokeStatic(query_processor.clj:538)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$process_query.invoke(query_processor.clj:529)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$dataset_query.invokeStatic(query_processor.clj:621)"
Nov 11 13:31:43 stdouterr.log:    "query_processor$dataset_query.invoke(query_processor.clj:579)"
Nov 11 13:31:43 stdouterr.log:    "api.dataset$fn__22582$fn__22585.invoke(dataset.clj:34)"
Nov 11 13:31:43 stdouterr.log:    "api.common.internal$wrap_catch_api_exceptions.invokeStatic(internal.clj:222)"
Nov 11 13:31:43 stdouterr.log:    "api.common.internal$wrap_catch_api_exceptions.invoke(internal.clj:219)"
Nov 11 13:31:43 stdouterr.log:    "api.dataset$fn__22582.invokeStatic(dataset.clj:28)"
Nov 11 13:31:43 stdouterr.log:    "api.dataset$fn__22582.invoke(dataset.clj:28)"
Nov 11 13:31:43 stdouterr.log:    "middleware$enforce_authentication$fn__30829.invoke(middleware.clj:84)"
Nov 11 13:31:43 stdouterr.log:    "api.routes$fn__30935.invokeStatic(routes.clj:43)"
Nov 11 13:31:43 stdouterr.log:    "api.routes$fn__30935.invoke(routes.clj:43)"
Nov 11 13:31:43 stdouterr.log:    "routes$fn__31914.invokeStatic(routes.clj:24)"
Nov 11 13:31:43 stdouterr.log:    "routes$fn__31914.invoke(routes.clj:24)"
Nov 11 13:31:43 stdouterr.log:    "middleware$log_api_call$fn__30917$fn__30919.invoke(middleware.clj:275)"
Nov 11 13:31:43 stdouterr.log:    "db$_do_with_call_counting.invokeStatic(db.clj:458)"
Nov 11 13:31:43 stdouterr.log:    "db$_do_with_call_counting.invoke(db.clj:452)"
Nov 11 13:31:43 stdouterr.log:    "middleware$log_api_call$fn__30917.invoke(middleware.clj:274)"
Nov 11 13:31:43 stdouterr.log:    "middleware$add_security_headers$fn__30876.invoke(middleware.clj:209)"
Nov 11 13:31:43 stdouterr.log:    "middleware$bind_current_user$fn__30833.invoke(middleware.clj:104)"],
Nov 11 13:31:43 stdouterr.log:   :query
Nov 11 13:31:43 stdouterr.log:   {:type "native",
Nov 11 13:31:43 stdouterr.log:    :native {:query "SELECT 1000::money", :collection "widgets", :template_tags {}},
Nov 11 13:31:43 stdouterr.log:    :parameters [],
Nov 11 13:31:43 stdouterr.log:    :constraints {:max-results 10000, :max-results-bare-rows 2000},
Nov 11 13:31:43 stdouterr.log:    :info {:executed-by 1, :card-id nil, :uuid "1ec9e5ea-98dc-4388-9015-9de9cd21fddb", :query-hash -1539598945, :query-type "native"}},
Nov 11 13:31:43 stdouterr.log:   :expanded-query nil}
Nov 11 13:31:43 stdouterr.log:
Nov 11 13:31:43 stdouterr.log:  11-11 20:31:43 ERROR metabase.query-processor :: Query failure: Bad value for type double : 1,000.00
Nov 11 13:31:43 stdouterr.log:  ["query_processor$assert_valid_query_result.invokeStatic(query_processor.clj:577)"
Nov 11 13:31:43 stdouterr.log:   "query_processor$assert_valid_query_result.invoke(query_processor.clj:572)"
Nov 11 13:31:43 stdouterr.log:   "query_processor$dataset_query.invokeStatic(query_processor.clj:622)"
Nov 11 13:31:43 stdouterr.log:   "query_processor$dataset_query.invoke(query_processor.clj:579)"
Nov 11 13:31:43 stdouterr.log:   "api.dataset$fn__22582$fn__22585.invoke(dataset.clj:34)"
Nov 11 13:31:43 stdouterr.log:   "api.common.internal$wrap_catch_api_exceptions.invokeStatic(internal.clj:222)"
Nov 11 13:31:43 stdouterr.log:   "api.common.internal$wrap_catch_api_exceptions.invoke(internal.clj:219)"
Nov 11 13:31:43 stdouterr.log:   "api.dataset$fn__22582.invokeStatic(dataset.clj:28)"
Nov 11 13:31:43 stdouterr.log:   "api.dataset$fn__22582.invoke(dataset.clj:28)"
Nov 11 13:31:43 stdouterr.log:   "middleware$enforce_authentication$fn__30829.invoke(middleware.clj:84)"
Nov 11 13:31:43 stdouterr.log:   "api.routes$fn__30935.invokeStatic(routes.clj:43)"
Nov 11 13:31:43 stdouterr.log:   "api.routes$fn__30935.invoke(routes.clj:43)"
Nov 11 13:31:43 stdouterr.log:   "routes$fn__31914.invokeStatic(routes.clj:24)"
Nov 11 13:31:43 stdouterr.log:   "routes$fn__31914.invoke(routes.clj:24)"
Nov 11 13:31:43 stdouterr.log:   "middleware$log_api_call$fn__30917$fn__30919.invoke(middleware.clj:275)"
Nov 11 13:31:43 stdouterr.log:   "db$_do_with_call_counting.invokeStatic(db.clj:458)"
Nov 11 13:31:43 stdouterr.log:   "db$_do_with_call_counting.invoke(db.clj:452)"
Nov 11 13:31:43 stdouterr.log:   "middleware$log_api_call$fn__30917.invoke(middleware.clj:274)"
Nov 11 13:31:43 stdouterr.log:   "middleware$add_security_headers$fn__30876.invoke(middleware.clj:209)"
Nov 11 13:31:43 stdouterr.log:   "middleware$bind_current_user$fn__30833.invoke(middleware.clj:104)"]
```
</details>

**Comment by camsaul** [comment]
It sounds like this is an issue with the Postgres JDBC driver, not with Metabase (it should probably be using something like `BigDecimal` for `MONEY` types instead of `Float`).

In fact it looks like they have an open issue here: https://github.com/pgjdbc/pgjdbc/issues/425

**Comment by camsaul** [comment]
Fixed by #11716
